packages/lib/src/vis/shared/SelectToZoom.tsx

```diff
@@ -49,13 +55,25 @@ function SelectToZoom(props: Props) {
   return (
     <SelectionTool modifierKey={modifierKey} onSelectionEnd={onSelectionEnd}>
       {({ startPoint, endPoint }) => (
-        <SelectionRect
-          startPoint={startPoint}
-          endPoint={endPoint}
-          fillOpacity={0.25}
-          fill="white"
-          stroke="black"
-        />
+        <>
+          <SelectionRect
+            startPoint={startPoint}
+            endPoint={endPoint}
+            fill="white"
+            stroke="black"
+            fillOpacity={keepRatio ? 0 : 0.25}
+            strokeDasharray={keepRatio ? '4' : undefined}
+          />
+          {keepRatio && (
+            <SelectionRect
+              startPoint={startPoint}
+              endPoint={getRatioEndPoint(startPoint, endPoint, width / height)}
+              fillOpacity={0.25}
+              fill="white"
+              stroke="black"
+            />
+          )}
+        </>
```

Comment:
The first `SelectionRect` shows the rectangle drawn by the user. The second after the `keepRatio` check shows the resulting FOV when the ratio is kept.

When `keepRatio` is `false`, the first rectangle already shows the resulting FOV so that we don't need to show the second.

This being said, I feel this part is not self-documenting enough. I should perhaps rename `endPoint` into `userEndPoint` or `pickedEndPoint` ?

Reply:
I think it's fine for now. It will perhaps be clearer once we start dealing the boundaries of the vis, as `getRatioEndPoint` may then have to accept and return a full selection rather than a single point (as the start point may also change). Depends on what we decide the behaviour should be, as we discussed via Zoom.